src/extension.ts

```diff
@@ -32,48 +32,45 @@ const COMMON_ERRORLOG_PATTERNS = [
     /log\.(Printf|Print|Println)\(['"]([^'"]+)['"]\)/gi,
 ]
 
-// TODO: Refactor to use activeEditor
 export function activate(context: sourcegraph.ExtensionContext): void {
-    sourcegraph.workspace.openedTextDocuments.subscribe(textDocument => {
-        const params: Params = getParamsFromUriPath(textDocument.uri)
-        const sentryProjects = SETTINGSCONFIG['sentry.projects']
+    if (sourcegraph.app.activeWindowChanges) {
+        const activeEditor = from(sourcegraph.app.activeWindowChanges).pipe(
+            filter((window): window is sourcegraph.Window => window !== undefined),
+            switchMap(window => window.activeViewComponentChanges),
+            filter((editor): editor is sourcegraph.CodeEditor => editor !== undefined)
+        )
+        // When the active editor changes, publish new decorations.
+        context.subscriptions.add(
+            activeEditor.subscribe(editor => {
+                const params: Params = getParamsFromUriPath(editor.document.uri)
+                const sentryProjects = SETTINGSCONFIG['sentry.projects']
 
-        // Retrieve the Sentry project that this document reports to.
-        // TODO: Move this outside of activate() and into a separate, testable function.
-        const sentryProject = sentryProjects && matchSentryProject(params, sentryProjects)
-        let missingConfigData: string[] = []
-        let fileMatched: boolean | null
+                // Retrieve the Sentry project that this document reports to.
+                // TODO: Move this outside of activate() and into a separate, testable function.
```

Comment:
Clarification for out-of-band question:

A separate, testable function could have the following signature:

```typescript
function getDecorations(document: TextDocument, settings: Settings): TextDocumentDecoration[]
```

It would basically help drastically reduce the amount of business logic present in the `activeEditor.subscribe()` callback:

```typescript
activeEditor.subscribe(editor => {
    const decorations = getDecorations(editor.document, settings)
    editor.setDecorations(decorationType, decorations)
})
```